### PR TITLE
If the media store favourite state is unknown, treat it as no favourite

### PR DESCRIPTION
**Pull request by Abestanis**
While adding tests to work towards #127, I noticed that we would crash if `isFavoriteInMediaStore` was `null`. We haven't seen this on Crashlytics yet, but this can definitely happen on older (pre `R`, API level `30`) devices.

### CODE REVIEW

File: lib/logic/player/favorites.dart [hunk 1]
```diff
@@ -54,7 +54,7 @@ class FavoritesControl with Control {
     final Set<int> favoriteSet = {};
     if (_useMediaStoreFavorites(contentType)) {
       for (final song in ContentControl.instance.state.allSongs.songs) {
-        if (song.isFavoriteInMediaStore!) {
```
Comment (nt4f04uNd) [inline lib/logic/player/favorites.dart:57]: This code must not be called on API <30

Currently it's filtered out by [this condition](https://github.com/nt4f04uNd/sweyer/blob/master/lib/logic/player/favorites.dart#L27), that's why there's a non-nullable assertion `!` there

If it would be called, I would like for this code to rather crash, than silently be executed

Comment (Abestanis) [reply lib/logic/player/favorites.dart:57]: `isFavoriteInMediaStore` is just an attribute that we get from the Android media database and we've seen that they can be null, also on API >=30. I really don't think we should allow our app to crash just because that value can be null, especially here during initialization which will prevent the whole app from loading up.

An alternative approach would be to change our model and make it not-nullable, and fall back to `false` if it is null in the media database.

Comment (nt4f04uNd) [reply lib/logic/player/favorites.dart:57]: I'd prefer for it to have a default value then, yeah

Comment (nt4f04uNd) [reply lib/logic/player/favorites.dart:57]: Actually, I see that on the Kotlin side on R and above we send `getInt() == 1`, so it will never be null in Dart

So technically, it can be null, but practically it currently will not

Still, putting a default value would be OK

Comment (Abestanis) [reply lib/logic/player/favorites.dart:57]: > Actually, I see that on the Kotlin side on R and above we send getInt() == 1, so it will never be null in Dart

Ah yes, indeed. But I agree, let's use a default value and make it not nullable.

**Review by nt4f04uNd — APPROVED** [review]
